src/scene/scene_lua.inl

```diff
@@ -28,73 +28,115 @@ namespace openspace {
+        // ownership list
+        if (tagMatchOwner == nullptr)
+            tagMatchOwner = findPropertyOwnerWithMatchingGroupTag(owner, tagToMatch);
+    }
+    return tagMatchOwner;
+}
+
+void applyRegularExpression(lua_State* L, std::regex regex,
+                            std::vector<properties::Property*> properties, int type,
+                            std::string groupName = "")                               {
     using ghoul::lua::errorLocation;
     using ghoul::lua::luaTypeToString;
-
+    bool isGroupMode = (groupName.empty()) ? false : true;
+    
     for (properties::Property* prop : properties) {
         // Check the regular expression for all properties
         std::string id = prop->fullyQualifiedIdentifier();
+
         if (std::regex_match(id, regex)) {
             // If the fully qualified id matches the regular expression, we queue the
             // value change if the types agree
-
+            if (isGroupMode) {
+                properties::PropertyOwner* matchingTaggedOwner = findPropertyOwnerWithMatchingGroupTag(prop,
+                groupName);
+                if (matchingTaggedOwner == nullptr)
+                    continue;
+            }
+            
             if (type != prop->typeLua()) {
                 LERRORC("property_setValue",
                         errorLocation(L) << "Property '" <<
                         prop->fullyQualifiedIdentifier() <<
                         "' does not accept input of type '" << luaTypeToString(type) <<
                         "'. Requested type: '" << luaTypeToString(prop->typeLua()) << "'"
                 );
+            } else {
+               executePropertySet(prop, L);
             }
-            else {
-                prop->setLuaValue(L);
-                //ensure properties are synced over parallel connection
-                std::string value;
-                prop->getStringValue(value);
-/*                OsEng.parallelConnection().scriptMessage(
-                    prop->fullyQualifiedIdentifier(),
-                    value
-                );*/
-            }
-
         }
     }
 }
-}
-
 
-namespace luascriptfunctions {
+bool doesUriContainGroupTag(const std::string command) {
+    std::string name = command.substr(0, command.find_first_of("."));
+    if (name.front() == '{' && name.back() == '}')
+        return true;
+    else
+        return false;
+}
 
-/**
- * \ingroup LuaScripts
- * setPropertyValueSingle(string, *):
- * Sets the property identified by the URI in the first argument to the value passed to
- * the second argument. The type of the second argument is arbitrary, but it must agree
- * with the type the denoted Property expects
- */
-int property_setValueSingle(lua_State* L) {
-    using ghoul::lua::errorLocation;
-    using ghoul::lua::luaTypeToString;
+std::string extractGroupNameFromUri(const std::string command) {
+    if (doesUriContainGroupTag(command)) {
```

Comment:
Yes this way would be smoother. I like my naming for things like "does URI contain" or "extract URI from". I initially broke up the work into functions and then after bringing everything together it becomes apparent that these all work together anyway. I'll make (partial?) changes on this tomorrow.